### ISSUE TITLE
Update harfbuzz port

### ISSUE THREAD

**Issue by ebraminio** [emscripten-core/emscripten#11668]
HarfBuzz is moving away from providing cmake port [used in emscripten to provide harfbuzz](https://github.com/emscripten-core/emscripten/blob/d58d738/tools/ports/harfbuzz.py#L35-L45), however, guess instead forcing the project to use the meson port (which would be fantastic if possible), emscripten can just use harfbuzz's native amalgam/unity one source support used also in [uharfbuzz](https://github.com/harfbuzz/uharfbuzz/commit/985eda07c7e7d9083b5403e2b67fbd1691fba7e8#diff-2eeaed663bd0d25b7e608891384b7298R30), src/harfbuzz.cc of harfbuzz repository, to get the library compiled in emscripten, guess just needs to pass a HAVE_PTHREAD and HAVE_FREETYPE to CC when compiling src/harfbuzz.cc (one can see what cmake already passes however), I will also try this when could setup emscripten's development environment.

tldr: `emcc src/harfbuzz.cc -DHAVE_PTHREAD -DHAVE_FREETYPE -fno-exceptions -fno-rtti -fvisibility-inlines-hidden` can be used instead all the things happened on the harfbuzz port here.

**Comment by kripken** [comment]
Interesting, thanks for the info!

PR would be great if you or someone else has time, adding labels.

**Comment by sbc100** [comment]
I think we should support meson .. I think some people have been using it already.   See https://github.com/emscripten-core/emscripten/issues/11066 for example

**Comment by sbc100** [comment]
Fixed in #14497
